src/search/lp/lp_solver.cc

```diff
@@ -304,6 +378,23 @@ void LPSolver::solve() {
     }
 }
 
+void LPSolver::write_lp(string filename) const {
+    try {
+        lp_solver->writeLp(filename.c_str());
+    } catch (CoinError &error) {
+        handle_coin_error(error);
+    }
+}
+
+void LPSolver::print_failure_analysis() const {
+    cout << "isAbandoned: " << lp_solver->isAbandoned() << endl;
```

Comment:
I'd change the strings from `isAbandoned` to `abandoned`, `isProvenOptimal` to `proven optimal` and so on.